src/docs/reference/core/Configuration/README.md

```diff
@@ -18,6 +18,9 @@ Orchard Core supports a hierarchy of Configuration Sources
 
 The Configuration Sources are loaded in the above order, and settings lower in the hierarchy will override values configured higher up, i.e. an Global Tenant value will always be overridden by an Environment Variable.
 
+!!! note 
+    Not all modules, even if they expose `IOptions`-based configuration, have built-in support for `IShellConfiguration` keys. So the patterns in the `appsettings.json` examples below won't necessarily work for just any module out of the box. Keep reading on how you can combine the two.
```

Comment:
```suggestion
    Modules that are typically configured through the admin with site settings do not expose their settings via `IShellConfiguration`. Keep reading for details on how you can combine site settings with `IShellConfiguration`.
```

Reply:
The confusion stems from, at least for me and the person under the issue I've linked, is that `IOptions` naturally brings the possibility of using ASP.NET Core configuration keys. I'd never assume that something can be configured from appsettings.json just because it has settings on the admin. Having `IOptions`-based settings, however, is halfway there, but only halfway, and that's what's confusing. This is what I want to clear up here.

Reply:
Ok, the issue that you linked was where someone specifically did think that it would be automatically configured from `appsettings.json` even though it was typically done in site settings, so that's what i am assuming you are trying to clear up here.

I'm not saying what you have written is wrong. 

I am just asking you to be specific, because when you say things like `Not all modules` and `won't necesaraily work for any module out of the box` it reads like we don't actually know which modules it works for, and which ones it doesn't

Reply:
I made this note simpler and added the second part of it closer to the example. Looks good?